### ISSUE TITLE
Dismiss notification by swiping

### ISSUE THREAD

**Issue by 2ntense** [kabouzeid/phonograph-issue-tracker#92]
I find this feature very useful when using Play Music. The notification should only be swipable when the music is stopped or paused.

**Comment by kabouzeid** [comment]
We had this in an earlier version of Phonograph. However it doesn't play well with other parts of the app when running on devices with low memory. If I can find a way to fix that then I will reimplement it.

**Comment by xiprox** [comment]
How about making it swipeable only when music is not playing? I wouldn't think there would be a problem with the app being cleared from memory then, would there?

**Comment by kabouzeid** [comment]
Exactly. The problem was when you open a lot of apps and play music, and then pause it phonograph gets killed by the system because the notification is no longer in foreground mode.

**Comment by 2ntense** [comment]
How about making it an option?

**Comment by okungnyo** [comment]
So? Then the user can just open Ponograph again.

Default behavior should be: when music is paused, let us swipe away Ponograph.

If low memory causes Ponograph to be killed, then the user can just open Ponograph again.

It takes two seconds. It's not a big deal.

**Comment by xiprox** [comment]
@kabouzeid Oh, I see.

@okungnyo Nope. What if the user just pauses it for a second there say, to be able to hear what someone calling out to them has to say? Then they try to resume and nothing happens. Your app is broken right there, and the user won't think anything else!

@kabouzeid Then again, what @2ntense said kind of makes sense. You could have it there as an option and warn the users about the issue they'll have if their phone has low memory. People with high-end phones would then be able to use this feature. Well, it's a whole different story if the killing still happens occasionally on high-end phones as well. But still, a warning would probably be fine in that case too.

**Comment by okungnyo** [comment]
What do you mean? Does the notification not disappear automatically when the app is killed due to low memory?

If it leaves behind a "phantom notification player" that doesn't respond, that is a separate bug that needs to be fixed.

**Comment by xiprox** [comment]
Oh, not sure about that. Then again, an app going away after pausing (with intent to resume right after) is a broken app. In that case, the pause button is no different than the `x` button. Do this and watch people complain on the Play Store listing page.

**Comment by okungnyo** [comment]
Actually yes, you are right. I would be furious if I pressed pause on anything and it disappeared on me for seemingly no reason.

So until low memory devices cease to exist (in five years time maybe lol?) the current implementation is a necessary evil, I guess.

**Comment by kabouzeid** [comment]
When the app gets killed, then the notification disappears as well.
